dali/test/python/test_operator_one_hot.py

```diff
@@ -15,39 +15,88 @@
 from nvidia.dali.pipeline import Pipeline
 import nvidia.dali.ops as ops
 import nvidia.dali.types as types
+from test_utils import check_batch
 
 import numpy as np
 
-sample_size = 20
+num_classes = 20
+batch_size = 10
+
+
+def insert_as_axis(target, value, axis, total_axes):
+    return target[0:axis] + (value,) + target[axis:total_axes]
 
 class OneHotPipeline(Pipeline):
-    def __init__(self, num_classes, input, num_threads=1):
-        super(OneHotPipeline, self).__init__(sample_size,
+    def __init__(self, num_classes, input, axis=-1, num_threads=1):
+        super(OneHotPipeline, self).__init__(batch_size,
                                              num_threads,
                                              0)
-        self.ext_src = ops.ExternalSource(source=[input], cycle=True)
-        self.one_hot = ops.OneHot(num_classes=num_classes, dtype=types.INT32, device="cpu")
+        sample_dim = len(input[0].shape)
+        self.ext_src = ops.ExternalSource(source=[input], cycle=True, layout="ABCD"[0:sample_dim])
+        self.one_hot = ops.OneHot(num_classes=num_classes, axis=axis, dtype=types.INT32, device="cpu")
 
     def define_graph(self):
         self.data = self.ext_src()
         return self.one_hot(self.data)
 
+def one_hot_3_axes(input, axis):
+    total_axes = len(input[0].shape)
+    assert total_axes == 3
+    axis = axis if axis >= 0 else total_axes
+    shapes = []
+    results = []
+    for i in range(batch_size):
+        shape = insert_as_axis(input[i].shape, num_classes, axis, total_axes)
+        result = np.zeros(shape, dtype=np.int32)
+        shapes.append(shape)
+        for i0 in range(input[i].shape[0]):
+            for i1 in range(input[i].shape[1]):
+                for i2 in range(input[i].shape[2]):
+                    in_coord = (i0, i1, i2)
+                    out_coord = insert_as_axis(in_coord, input[i][in_coord], axis, total_axes)
```

Comment:
I don't think this is how TF oneHot works. Now it is multiple hot. Can you provide rationale/reference for such behavior?

Reply:
```
If indices is a matrix (batch) with shape [batch, features], the output shape will be:

  batch x features x depth if axis == -1
  batch x depth x features if axis == 1
  depth x batch x features if axis == 0
```
from [tf.one_hot doc](https://www.tensorflow.org/api_docs/python/tf/one_hot)